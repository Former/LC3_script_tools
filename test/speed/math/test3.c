
int main()
{
    int k, i, res = 0;

    for (k = 1; k < 30; ++k)
    {
        for (i = 0; i < 5000; ++i)
        {
            res += (i >> 1 + k << 2) & 0xFF + i % k + i / k;
            res &= 0x1F;
        }
        printf("%d res = %d\n", k, res);
   }

    printf("Test finish %d\n", res);
    return 0;
}
