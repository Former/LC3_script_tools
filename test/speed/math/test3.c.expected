1 res = 12
2 res = 12
3 res = 12
4 res = 28
5 res = 20
6 res = 4
7 res = 24
8 res = 8
9 res = 8
10 res = 16
11 res = 4
12 res = 4
13 res = 4
14 res = 4
15 res = 4
16 res = 4
17 res = 4
18 res = 4
19 res = 4
20 res = 4
21 res = 4
22 res = 4
23 res = 4
24 res = 4
25 res = 4
26 res = 4
27 res = 4
28 res = 4
29 res = 4
Test finish 4
